Closing JASP while bootstrapping makes it crash

boutinb opened the pull request:
Fixes jasp-stats/jasp-test-release#826

. Make the Analysis object as child of the AnalysisForm
. Do not destroy the Analysis object from Analyses: as child of
AnalysisForm, it will be destroyed automatically.
. When the AnalysisForm starts to be destroyed, signal it to all
objects that have a pointer to an analysis (EngineRepresentation and
PlotEditorModel), so that this pointer should not be used anymore.

JorisGoosen commented:
I'll just download your branch and play around with it, because this really needs to be checked well.

JorisGoosen commented:
But that'll be tomorrow.

boutinb commented:
I think this is OK for 0.13, but this is not a Must Have. So if you find it too dangerous, let's skip it.

AlexanderLyNL commented:
Let's skip this for now, I'll move this to 0.14.

JorisGoosen commented:
Yeah that is for the best I think.

boutinb committed:
Closing JASP while bootstrapping makes it crash

Fixes jasp-stats/INTERNAL-jasp#953

. Make the Analysis object as child of the AnalysisForm
. Do not destroy the Analysis object from Analyses: as child of
AnalysisForm, it will be destroyed automatically.
. When the AnalysisForm starts to be destroyed, signal it to all
objects that have a pointer to an analysis (EngineRepresentation and
PlotEditorModel), so that this pointer should not be used anymore.

boutinb commented:
I have checked again this PR. I think this is OK now

JorisGoosen commented:
What happens if we switch to loaders for AnalysisForm though?
Then the Analysis gets destroyed whenever the form goes out of view.

boutinb commented:
The problem is not reproduceable anymore, maybe due to https://github.com/jasp-stats/jasp-desktop/commit/f7993fc2ec9d9d0fd5560a67e43a882bd3ebc578
And this fix would prevent https://github.com/jasp-stats/INTERNAL-jasp/issues/737 being implemented
